lib/cldr/helpers/map.ex

```diff
@@ -81,7 +81,7 @@ defmodule Cldr.Map do
   ## Examples
 
   """
-  @spec deep_map(Map.t(), key_function :: function(), value_function :: function()) :: Map.t()
+  @spec deep_map(term(), key_function :: function(), value_function :: function()) :: term()
```

Comment:
That does seem like a quite generous loosening of typespec. What's it actually called with that `map()` would be to strict?

Reply:
Because the implementation is recursive on `deep_map` the spec needs to reflect all the public clauses:

Line 89: `t:struct/0`
Line 93: `t:map/0`
Line 104: `t:list/0`
Line 108: `nil`
Line 112: `t:term/0` because it is the value in a map which can be any `term`.

Since `t:term/0` is the supertype of all of the other types and the `value_fun` can pass `value` through unchanged, the input and output types become `term`.  If you want to restrict the type to `map()`, you'll need to change the implementation to call a separate helper function that does the recursion so that `deep_map` will only accept maps.  Would you like me to make that change?

So you're aware, there was a call in the code that passed a list to `deep_map`.  It did look weird, but I was trying to not change code, just the specs whenever possible.

If you want me to fix that caller that used a list and/or make `deep_map` only accept a `map` by adding a separate private recursion helper, let me know.